### ISSUE TITLE
Support for > 2 hosts via PFX configured with > 2 NT partitions

### ISSUE THREAD

**Issue by eswierk** [Microsemi/switchtec-kernel#17]
PFX switches support from 12 to 48 NTB ports, potentially allowing 12-48 hosts with root-only (not endpoint) ports to communicate among each other peer-to-peer.

According to the switchtec-kernel README, the ntb_hw_switchtec driver "only supports switches configured with exactly 2 NT partitions and zero or more non-NT partitions", which implies limiting the switch to allow communication between at most 2 hosts.

Is this purely a limitation of the ntb_hw_switchtec driver, the ntb infrastructure in the Linux kernel, or something else?

What would be the most straightforward way to use a PFX switch for communication among more than 2 hosts with root-only ports? Is there any work underway to support such an application in Linux, by Microsemi or others?

**Comment by lsgunth** [comment]
When the driver was originally written the NTB framework in Linux only supported a two ports. 

Some work has been done to support multiple ports but it is still not complete and the ntb_hw_switchtec driver has not been updated to align with these changes. 

Also, the only NTB clients that support multiple hosts are the test clients. And it's a very difficult problem to add multi-host support to ntb_transport/ntb_netdev so until then multi-host support in ntb_hw_switchtec is a bit of a moot point.

Logan

**Comment by lsgunth** [comment]
No, I don't think anyone is working on it publicly. I know one company that has kind of done what you describe but it's part of a larger project. More over the upstream kernel community has gone in the other direction and have expanded the API to include multiple hosts but no one has added support for ntb_transport or any other client that would be useful. The switchtec NTB drive also hasn't been updated to support multiple hosts.

**Comment by chairwa** [comment]
Has there been any progress towards this feature?

**Comment by Kendidi** [comment]
Please share info. if there is any update on this multi-partitions support.  Many thanks in advance.
